Kitodo/src/main/java/org/kitodo/services/file/FileService.java

```diff
@@ -215,24 +248,35 @@ public URI renameFile(URI fileUri, String newFileName) throws IOException {
      */
     public Integer getNumberOfFiles(URI directory) {
         int count = 0;
-        if (isDirectory(directory)) {
-            /*
-             * die Unterverzeichnisse durchlaufen
-             */
+        if (directory.isAbsolute()) {
+            if (isDirectory(directory)) {
+                ArrayList<URI> children = getSubUris(directory);
+                for (URI aChildren : children) {
+                    if (isDirectory(aChildren)) {
+                        count += getNumberOfFiles(aChildren);
+                    } else {
+                        count += 1;
+                    }
+                }
+            }
+        } else {
+            directory = getAbsoluteURI(directory, MappingType.DATA, null, null);
             ArrayList<URI> children = getSubUris(directory);
-            for (URI aChildren : children) {
-                if (isDirectory(aChildren)) {
-                    count += getNumberOfFiles(aChildren);
-                } else {
-                    count += 1;
+            if (isDirectory(directory)) {
+                for (URI aChildren : children) {
+                    if (isDirectory(aChildren)) {
+                        count += getNumberOfFiles(aChildren);
+                    } else {
+                        count += 1;
+                    }
```

Comment:
Both iterations through the list of URIs looks similar: why did you not move this counting to its own method? This would even reduce the level of nested `if` statements.